source/base/parameter_handler.cc

```diff
@@ -1862,6 +1862,7 @@ namespace
 
 bool ParameterHandler::read_input_from_xml (std::istream &in)
 {
+  AssertThrow(in, ExcIO());
```

Comment:
To be fair, most of the functions will properly throw exceptions in addition to returning an exit status. Are you in favor of deprecating the functions that return `bool`s? Do we know how many people actually use the return values?